distribute/distribute.sh

```diff
@@ -1,193 +1,288 @@
-WINDOWS_JRE_URL=${WINDOWS_JRE_URL:-"https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.5%2B8/OpenJDK17U-jre_x64_windows_hotspot_17.0.5_8.zip"}
-WINDOWS_JRE_SHA256=${WINDOWS_JRE_SHA256:-"d25a2d44c1bd9c57d49c5e98de274cd40970ab057fe304b52eb459de4ee5d8a5"}
-
-if which makensis > /dev/null
-then
-	mkdir -p cache
-	echo "Downloading OpenJRE for Windows (may be overridden with WINDOWS_JRE_URL and WINDOWS_JRE_SHA256)"
-	# download JRE if not existing or wrong file contents
-    check_sha256 cache/jre.zip $WINDOWS_JRE_SHA256 || wget "$WINDOWS_JRE_URL" -O cache/jre.zip
-    # check if downloaded JRE is correct (to exclude incomplete download)
-    check_sha256 cache/jre.zip $WINDOWS_JRE_SHA256 || exit 1
-    echo "Building Windows launcher and installer"
-	# Copy files to wintmp/
-	rm -rf wintmp
-	mkdir wintmp
-	cp -r windows/* wintmp/
-	[ -d wintmp/stream ] || mkdir wintmp/stream
-	cp -r visicut/* wintmp/stream/
-	cp ../src/main/resources/de/thomas_oster/visicut/gui/resources/splash*.png wintmp/stream
-	cp ../tools/inkscape_extension/* wintmp/
-
-	# build setup.exe installer
-	# and VisiCut.exe launcher executable
-	cat windows/installer.nsi|sed s#VISICUTVERSION#"$VERSION"#g > wintmp/installer.nsi
-	pushd wintmp > /dev/null
-	# Unpack JRE. We assume that this creates a subfolder with "jre" in its name
-	unzip -q ../cache/jre.zip
-    mv *jre*/ stream/jre/
-	test -e stream/jre/bin/java.exe || { echo "Cannot find java.exe in JRE ZIP file"; exit 1; }
-	test -e stream/jre/legal/java.base/LICENSE || { echo "Cannot find license information in JRE ZIP file"; exit 1; }
-	# Build license text for Windows installer
-    {
-        cat ../../LICENSE
-        echo "Java Runtime for Windows:"
-        echo -e "- OpenJRE (GPLv2 with Classpath exception). License details follow:"
-        find stream/jre/legal/ -type f -print -exec cat '{}' ';'
-    }  > stream/license-with-jre.txt
-	makensis launcher.nsi > /dev/null || exit 1 # build VisiCut.exe
-	cp VisiCut.exe ../visicut/ # copy VisiCut.exe so that it is included in the platform independent ZIP
-	mv VisiCut.exe ./stream/
-	makensis installer.nsi > /dev/null || exit 1 # build setup.exe
-	popd
-	mv wintmp/setup.exe VisiCut-$VERSION-Windows-Installer.exe || exit 1
-	zip VisiCut-$VERSION-Windows-Installer.zip VisiCut-$VERSION-Windows-Installer.exe	# for github upload
+# targets to be built need to be passed on the commandline
+# if no targets are provided, we display a help text
+if [[ "${1:-}" == "" ]]; then
+    log "Usage: [env BUILD=1] $0 <targets>"
+    echo
+    log "Available targets:"
+    log "    - zip"
+    log "    - windows-nsis"
+    log "    - macos-bundle"
+    log "    - linux-appimage"
+    log "    - linux-checkinstall"
+    log "    - linux-makepkg"
+    echo
+    log "Available environment variables:"
+    log "    - \$BUILD=[...]: if set to any string, $0 will build VisiCut's JAR first (not set by default)"
```

Comment:
I'm not so happy about not building by default. The docker-build.sh also builds by default, and the previous distribute.sh also did it.

It simplifies usability if there is just one "do everything" script that builds a complete package by default.

Reply:
We can change the default and have users set a `NO_BUILD=1` or something like that.